internal/store/plugin.go

```diff
@@ -1,26 +1,60 @@
 package store
 
 import (
+	"strings"
+
 	"github.com/mattermost/mattermost-marketplace/internal/model"
 )
 
+func pluginMatchesFilter(plugin *model.Plugin, filter string) bool {
+	filter = strings.ToLower(filter)
+	if strings.ToLower(plugin.Manifest.Id) == filter {
+		return true
+	}
+
+	if strings.Contains(strings.ToLower(plugin.Manifest.Name), filter) {
+		return true
+	}
+
+	if strings.Contains(strings.ToLower(plugin.Manifest.Description), filter) {
+		return true
+	}
+
+	return false
+}
+
 // GetPlugins fetches the given page of plugins. The first page is 0.
-func (store *Store) GetPlugins(filter *model.PluginFilter) ([]*model.Plugin, error) {
-	if len(store.plugins) == 0 || filter.PerPage == 0 {
+func (store *Store) GetPlugins(pluginFilter *model.PluginFilter) ([]*model.Plugin, error) {
+	var plugins []*model.Plugin
+
+	filter := strings.TrimSpace(pluginFilter.Filter)
+	if filter == "" {
+		plugins = store.plugins
+	} else {
+		for _, plugin := range store.plugins {
+			if !pluginMatchesFilter(plugin, filter) {
+				continue
+			}
+
+			plugins = append(plugins, plugin)
+		}
+	}
+
+	if len(plugins) == 0 || pluginFilter.PerPage == 0 {
```

Comment:
Should we move this to the top before filtering? Just in case of the very rare case it requests 0 per page, it does not do an unnecessary pass through all the plugins to filter to later not return any